File: array-hashing/longest-common-prefix.go
package main

import (
	"fmt"
	"sort"
)

func LongestCommonPrefixDemo() {
	strs := []string{"flower", "flow", "flight"}
	result := LongestCommonPrefix(strs)
	fmt.Println(result)
}

func LongestCommonPrefix(strs []string) string {
	sort.Slice(strs, func(i, j int) bool {
		return len(strs[i]) < len(strs[j])
	})
	prefix := strs[0]
	prefix_length := len(prefix)

	for _, s := range strs[1:] {
		for prefix != s[0:prefix_length] {
			prefix_length -= 1
			if prefix_length == 0 {
				return ""
			}
			prefix = prefix[0:prefix_length]
		}
	}

	return prefix
}

File: array-hashing/longest-common-prefix.py
from typing import List

class Solution:
  @staticmethod
  def longestCommonPrefix(strs: List[str]) -> str:
    prefix: str = strs[0]
    prefix_length: int = len(prefix)
    
    for s in strs[1:]:
      while prefix != s[0:prefix_length]:
        prefix_length -= 1
        if prefix_length == 0:
          return ""
        prefix = prefix[0:prefix_length]
    
    return prefix
  
  
result =  Solution.longestCommonPrefix(["flower","flow","flight"])

print(result)

s = "flow"
a = s[0: 6]
print(a)
print(len(a))